plugins/lookup/laps.py

```diff
@@ -0,0 +1,233 @@
+# Copyright: (c) 2024, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+DOCUMENTATION = r"""
+name: laps
+author: Nico Ohnezat (@no-12)
+short_description: Inventory plugin for Active Directory
+version_added: 2.2.0
+description:
+- Lookup plugin that retrieves the LAPS password information for multiple hosts from an Active Directory server.
+options:
+  _terms:
+    description:
+    - One or multiple C(CommonName) of the computer objects to search for.
+    required: True
+  laps:
+    description:
+    - The LAPS password type to retrieve.
+    - Defaults to the C(auto).
+    - This will attempt to retrieve the LAPS password in the following order C(windows_encrypted), C(windows_plain_text), C(legacy_microsoft).
+    type: str
+    choices:
+    - auto
+    - windows_encrypted
+    - windows_plain_text
+    - legacy_microsoft
+    default: auto
+  search_base:
+    description:
+    - The LDAP search base to find the computer objects in.
+    - Defaults to the C(defaultNamingContext) of the Active Directory server
+      if not specified.
+    - If searching a larger Active Directory database, it is recommended to
+      narrow the search base to speed up the queries.
+    type: str
+  search_scope:
+    description:
+    - The scope of the LDAP search to perform.
+    - C(base) will search only the current path or object specified by
+      I(search_base). This is typically not useful for inventory plugins.
+    - C(one_level) will search only the immediate child objects in
+      I(search_base).
+    - C(subtree) will search the immediate child objects and any nested
+      objects in I(search_base).
+    choices:
+    - base
+    - one_level
+    - subtree
+    default: subtree
+    type: str
+notes:
+- This plugin is a tech preview and the module options are subject to change
```

Comment:
We should add a note that states some of the efficiency issues I've mentioned in the PR like

+ The lookup is run per task unless they define it through `set_fact`
+ Each lookup makes its own connection rather than sharing them
+ It is recommended to use the `microsoft.ad.ldap` plugin instead of this lookup.

Happy for you to put it in your own words.

For the last point you can add a plugin reference with `P(...)` like.

```yaml
- Some message P(microsoft.ad.ldap#lookup)
```